### PR TITLE
Feature/rb 56 check deads

### PR DESCRIPTION
**Pull request by kw4ntiq**
* **Please check if the PR fulfills these requirements**
- [x] The commit message follows our guidelines
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)
- [x] Norm of the code has been respected

* **In what state is your pull request?**
- [x] Ready to merge / Waiting a reviwer to see my work.
- [ ] Work In Progress (WIP) / My work is not finish but i want daily reviews. (Draft)
- [ ] CI Review / I want to see what the CI thinks of my work.
- [ ] Need feedback / Just to have feedback on what i produced. (May not be merged)

* **What kind of change does this PR introduce?** (You can choose multiple)
- [ ] Bug fix
- [ ] Feature request
- [x] New / Updated documentation
- [ ] Testing CI ( Make the pull request in draft mode)

* **What is the current behavior?** (link an issue based on the kind of change this pr introduce)

* **What is the new behavior (if this is a feature change)?**

* **Other information**:

### CODE REVIEW

File: src/Client/Systems/Graphic/DeathSystems.cpp [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+#include <unordered_map>
+#include "CustomTypes.hpp"
+#include "Registry.hpp"
+
+namespace Systems {
+
+    // REPLACE BY YOUR DEATH FUNCTION
+
+    const std::function<void(std::size_t)> testFuncPlayer = [](std::size_t id) {
+        std::cout << "Player " << id << " is dead" << std::endl;
```
Comment (guillaumeAbel) [inline src/Client/Systems/Graphic/DeathSystems.cpp]: set animRect to dead rect

**Review by KitetsuK — APPROVED** [review]

File: src/Client/Systems/Graphic/DeathSystems.cpp [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+        deathFunctions = {
+            {std::type_index(typeid(Types::Player)), testFuncPlayer},
+    };
+
+    static void addDeathFunction(
+        Registry::components<Types::Dead> &arrDead,
+        Registry::components<Types::Player> &arrPlayer,
+        std::size_t id)
+    {
+        try {
```
Comment (guillaumeAbel) [inline src/Client/Systems/Graphic/DeathSystems.cpp]: should iterate over a vector of type_index

File: src/ECS/Systems/Systems.cpp [hunk 3]
```diff
@@ -101,26 +101,46 @@ namespace Systems {
...
+        Registry::components<Types::Health> arrHealth =
+            Registry::getInstance().getComponents<Types::Health>();
+        Registry::components<Types::Dead> arrDead =
+            Registry::getInstance().getComponents<Types::Dead>();
+
+        std::vector<std::size_t> ids = arrHealth.getExistingsId();
+
+        for (auto itIds = ids.begin(); itIds != ids.end(); itIds++) {
+            if (arrHealth[*itIds].hp <= 0 && arrDead.exist(*itIds)) {
+                arrDead[*itIds].deathFunction
```
Comment (guillaumeAbel) [inline src/ECS/Systems/Systems.cpp]: call static function

**Review by guillaumeAbel — CHANGES_REQUESTED** [review]

File: src/Client/SceneManager.cpp [hunk 4]
```diff
@@ -14,8 +14,8 @@
 
 #include "CustomTypes.hpp"
 
-constexpr int screenWidth  = 1920;
-constexpr int screenHeight = 1080;
+constexpr int screenWidth  = 800;
```
Comment (Saverio976) [inline src/Client/SceneManager.cpp]: Huh ?

File: CMakeLists.txt [hunk 5]
```diff
@@ -56,25 +56,7 @@ install(TARGETS ${PROJECT_NAME_SERVER}
...
-        )
-        set_target_properties(${PROJECT_NAME_SERVER} PROPERTIES
-            VS_GLOBAL_RunCodeAnalysis true
-            VS_GLOBAL_EnableClangTidyCodeAnalysis true
-        )
-    else()
-        set(CLANG_TIDY_COMMAND "${CLANG_TIDY_EXE}" "--fix" "--fix-notes" "--fix-errors")
-        set(CMAKE_CXX_CLANG_TIDY "${CLANG_TIDY_COMMAND}")
-    endif()
-endif()
```
Comment (Saverio976) [inline CMakeLists.txt]: Pourquoi ?

**Review by Saverio976 — CHANGES_REQUESTED** [review]

File: src/ECS/ECSCustomTypes.hpp [hunk 6]
```diff
@@ -7,7 +7,9 @@
 
 #pragma once
 
-// all values are in percentage of the screen
```
Comment (Saverio976) [inline src/ECS/ECSCustomTypes.hpp:10]: Pourquoi supprimer ?

**Review by guillaumeAbel — APPROVED** [review]

**Review by Saverio976 — APPROVED** [review]

**Pull request merged**